### ISSUE TITLE
signing image always fail with multuple stage docker build

### ISSUE THREAD

**Issue by LeoQuote** [sigstore/cosign-installer#47]
**Description**

recently I added a golang docker build for a project, but it always fail in signing process, error log:

```
Run cosign sign ghcr.io/LeoQuote/custom-error-page@sha256:f48b41f7085b07cfbd5fde9f56712aa08971caf858bf3a03eaed0f366650c566
Generating ephemeral keys...
Retrieving signed certificate...
client.go:194: root pinning is not supported in Spec 1.0.19
Successfully verified SCT...
Error: signing [ghcr.io/LeoQuote/custom-error-page@sha256:f48b41f7085b07cfbd5fde9f56712aa08971caf858bf3a03eaed0f366650c566]: parsing reference: could not parse reference: ghcr.io/LeoQuote/custom-error-page@sha256:f48b41f7085b07cfbd5fde9f56712aa08971caf858bf3a03eaed0f366650c566
main.go:46: error during command execution: signing [ghcr.io/LeoQuote/custom-error-page@sha256:f48b41f7085b07cfbd5fde9f56712aa08971caf858bf3a03eaed0f366650c566]: parsing reference: could not parse reference: ghcr.io/LeoQuote/custom-error-page@sha256:f48b41f7085b07cfbd5fde9f56712aa08971caf858bf3a03eaed0f366650c566
Error: Process completed with exit code 1.
```

here's the project using the action : https://github.com/LeoQuote/custom-error-page/pull/1/files

everything else seems to be fine. the image has been push, except image signing

**Comment by developer-guy** [comment]
hi @LeoQuote, I couldn't reproduce the issue, everything worked fine for me in my fork of your repository, see.

https://github.com/developer-guy/custom-error-page/runs/4739269964?check_suite_focus=true

can you please verify, thanks in advance?

**Comment by LeoQuote** [comment]
Is that possible that the real thing that trigger this problem is that the image was pushed to ghcr.io/leoquote/custom-error-page:main and the action is trying to sign image ghcr.io/LeoQuote/custom-error-page:main ？

notice the capital L and Q

you didn’t trigger this bug cause you don’t have capital character in your username .

Is that possible to sign ghcr.io/leoquote/custom-error-page:main to see if it can succeed?

**Comment by LeoQuote** [comment]
I ve noticed this wiki in docker-build-push action https://github.com/docker/build-push-action/blob/master/TROUBLESHOOTING.md saying “repository name must be lowercase”

Either by a standalone step to get lowercase repo name, or using docker meta action to achieve it.

**Comment by Jleagle** [comment]
> I ve noticed this wiki in docker-build-push action https://github.com/docker/build-push-action/blob/master/TROUBLESHOOTING.md saying “repository name must be lowercase”
> 
> Either by a standalone step to get lowercase repo name, or using docker meta action to achieve it.

This was it for me, thanks. The error does not match the issue.

**Comment by bobcallaway** [comment]
I moved up to the latest cosign-installer and cosign versions, and things went through fine on a quick fork of your project. Is there a reason you need to use v1.4.0?

https://github.com/bobcallaway/custom-error-page/runs/4794440126?check_suite_focus=true

**Comment by xNinjaKittyx** [comment]
I'm also running into this problem. I'm pretty sure the issue is regarding the fact that we have capital letters in our usernames. And would probably explain why it would always work if someone with a lowercase username forks a project.

I tried using 1.4.1 and that didn't work either for me.

**Comment by bobcallaway** [comment]
Thanks @xNinjaKittyx - I changed my repo name to have capital letters, and reproduced the originally reported problem. I also confirmed your solution to use works for me:

`cosign sign ${{ steps.meta.outputs.tags }}@${{ steps.build-and-push.outputs.digest }}`

https://github.com/bobcallaway/Custom-Error-Page/runs/5020316291?check_suite_focus=true

Based on that, I'm inclined to close this bug (as the issue is not with installing cosign into the runner), and open an issue in https://github.com/sigstore/cosign that suggests we look for non-lowercase characters in the repository path and log a more useful error message. Any objections?
